codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderConstraintViolationsTest.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.rust.codegen.core.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverIntegrationTest
+
+class ServerBuilderConstraintViolationsTest {
+
+    @Test
+    fun `it should not generate constraint violations for members with a default value`() {
+        val model = """
```

Comment:
Nice. I'd leave a comment linking to https://github.com/awslabs/smithy-rs/issues/2343 so the reader can easily find how this test came to be.

In general I like how the Rust compiler adds test for reported issues and links back to the info on GitHub e.g. https://github.com/rust-lang/rust/tree/master/tests/ui/codegen